parser/LesBonsNumerosAPIParser.go

```diff
@@ -0,0 +1,193 @@
+package parser
+
+import (
+	l "FDJ_SLACK/log"
+	"FDJ_SLACK/model"
+	"FDJ_SLACK/utils"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/etree"
+)
+
+type LesBonsNumerosAPIParser struct {
+}
+
+func NewParser() *LesBonsNumerosAPIParser {
+	s := &LesBonsNumerosAPIParser{}
+	return s
+}
+
+func (p *LesBonsNumerosAPIParser) GetAndParseData(apiURL string) []model.LotteryResult {
+	l.Info("Get data from ApiURL ", apiURL)
+
+	//Get data from URL
+	response, err := http.Get(apiURL)
+	if err != nil {
+		l.Error("%s", err)
+		os.Exit(1)
+	}
+
+	defer response.Body.Close()
+	contents, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		l.Error("%s", err)
+		os.Exit(1)
+	}
+
+	l.Info("Parsing data")
+
+	//Create new XML Document to go through results
+	doc := etree.NewDocument()
+	if err := doc.ReadFromBytes(contents); err != nil {
+		panic(err)
+	}
+
+	var result []model.LotteryResult
+
+	//For each item (using XPath to get description)
+	for _, t := range doc.FindElements("//item/description") {
+		l.Info("Found item")
+		for _, child := range t.Child {
+			if c, ok := child.(*etree.CharData); ok {
+
+				// Manage only not empty strings
+				if !isWhitespace(c.Data) {
+
+					//Cleaning Result to remove any HTML tags (ex: <h1>)
+					cleanHTML := utils.CleanHTML(c.Data)
+
+					//Get result line by line
+					resultByLine := strings.Split(cleanHTML, "\n")
+
+					l.Info(resultByLine)
+
+					date := extractDateLottery(resultByLine[1])
+					balls := extractBalls(resultByLine[2])
+					luckyBall := extractLuckyBall(resultByLine[3])
+					numberWinner := 0
```

Comment:
define defaults values in the LotteryResult struct instead of initilization in this method